e2e/internal/e2e/profile.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package e2e
```

Comment:
the code feels like it should be in a separate package.

Reply:
Because from what I can see this is very tightly related functionality within the new code and is not as tightly related to the rest of the existing code in the package. Sure, there's a high level theme here (e2e) but all the new code exists for a specific purpose.

Keep in mind that in Go the name of the package becomes part of the client code. Consider the `io` package and its `Reader` type. In the client code you have `io.Reader`, which is very expressive. The `io` package provides functionality that is closely related to `sys` and `os`. They could have provided everything in the `os` package (and that was the partly the case a long time ago) and have `os.Reader`. What's an `os.Reader`? To clarify this, they would have named the `Reader` type `IOReader` or `FileReader` or something like that (and I believe that or something similar was in fact the name at some point).

What I'm saying for example is what's better from the point of view of someone reading the code: `e2e.Match` or `profile.Match`?

Reply:
the `profile.Match` is better IMO, and actually calls are always done with `profile.Match` or depending of "profile" variable name, since `Match` is a function of `SingularityProfile`, so it's fine to keep it as is actually